sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/ExecutorRequirement.java

```diff
@@ -55,20 +63,23 @@ private static ExecutorRequirement getExistingExecutorRequirement(ExecutorInfo e
         }
     }
 
-    private ExecutorRequirement(ExecutorInfo executorInfo)
-            throws InvalidRequirementException {
+    private ExecutorRequirement(
+            ExecutorInfo executorInfo,
+            Collection<ResourceRequirement> resourceRequirements) throws InvalidRequirementException {
         validateExecutorInfo(executorInfo);
         this.executorInfo = executorInfo;
+        this.resourceRequirements = resourceRequirements != null ? resourceRequirements :
```

Comment:
It's not really possible to move it over to exclusively instantiating resource requirements from the resources themselves -- since subtypes of `ResourceRequirement` (e.g. `VolumeRequirement`) are polymorphic in the `ResourceEvaluationStage` they yield, we need to keep a handle to whichever `ResourceRequirement` we got from the specification. The ability to not pass in resource requirements explicitly was useful in the case of an existing offer requirement, which seemed safe since disk resources can't change after creation (so we don't need the volume-create logic in `VolumeEvaluationStage` because we intend to just claim the volume that we already reserved as-is), but of course I realize now that a user might want to add a new volume entirely to the pod. So probably the best bet is to always pass in the `ResourceRequirement` collection.